/* -------------------------------------------------------------------------- */
/* Copyright 2002-2025, OpenNebula Project, OpenNebula Systems                */
/*                                                                            */
/* Licensed under the Apache License, Version 2.0 (the "License"); you may    */
/* not use this file except in compliance with the License. You may obtain    */
/* a copy of the License at                                                   */
/*                                                                            */
/* http://www.apache.org/licenses/LICENSE-2.0                                 */
/*                                                                            */
/* Unless required by applicable law or agreed to in writing, software        */
/* distributed under the License is distributed on an "AS IS" BASIS,          */
/* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   */
/* See the License for the specific language governing permissions and        */
/* limitations under the License.                                             */
/* -------------------------------------------------------------------------- */
syntax = "proto3";

import "shared.proto";

package one.host;

option go_package = "github.com/OpenNebula/one/src/oca/go/src/goca/api/host";

message AllocateRequest
{
  string session_id = 1;
  string name       = 2;
  string im_mad     = 3;
  string vm_mad     = 4;
  sint32 cluster_id = 5;
}

message DeleteRequest
{
  string session_id = 1;
  int32 oid         = 2;
}

message InfoRequest
{
  string session_id = 1;
  int32 oid         = 2;
  bool decrypt      = 3;
}

message UpdateRequest
{
  string session_id = 1;
  int32 oid         = 2;
  string template   = 3;
  int32 append      = 4;
}

message RenameRequest
{
  string session_id = 1;
  int32 oid         = 2;
  string name       = 3;
}

message StatusRequest
{
  string session_id = 1;
  int32 oid         = 2;
  int32 status      = 3;
}

message MonitoringRequest
{
  string session_id = 1;
  int32 oid         = 2;
}

message PoolInfoRequest
{
  string session_id = 1;
}

message PoolMonitoringRequest
{
  string session_id = 1;
  int32 seconds = 2;
}

service HostService
{
  rpc Allocate (one.host.AllocateRequest) returns (one.ResponseID);

  rpc Delete (one.host.DeleteRequest) returns (one.ResponseID);

  rpc Info (one.host.InfoRequest) returns (one.ResponseXML);

  rpc Update (one.host.UpdateRequest) returns (one.ResponseID);

  rpc Rename (one.host.RenameRequest) returns (one.ResponseID);

  rpc Status (one.host.StatusRequest) returns (one.ResponseID);

  rpc Monitoring (one.host.MonitoringRequest) returns (one.ResponseXML);

  rpc PoolInfo (one.host.PoolInfoRequest) returns (one.ResponseXML);

  rpc PoolMonitoring (one.host.PoolMonitoringRequest) returns (one.ResponseXML);
}
